src/stdlib_io.f90

```diff
@@ -0,0 +1,104 @@
+module stdlib_io
+use stdlib_types
```

Comment:
Question: Why using stdlib_types, instead of iso_fortran_env?

Reply:
Yeah, we need to have a discussion about this and figure out how to provide the `dp`, `sp`, ... types. Let's move this into a separate issue #25, because we will be tackling this at every PR.

From that issue I am using the options 1a in this PR and you are asking why not to use 2b instead, am I right? That's a valid question and let's discuss it at #25.

Reply:
You are right! See answer in #25

Reply:
Ok, so we agreed in #25 to remove `stdlib_types` and use `iso_fortran_env` directly in each module that needs it for now as follows:
```fortran
use iso_fortran_env, only: sp=>real32, dp=>real64, qp=>real128
```
and the actual variable names will stay as `sp`, `dp` and `qp`, and they will only be used internally by `stdlib` by the nature of using `iso_fortran_env` in each module explicitly as shown above.

It's a compromise that should allow us to move forward. I'll make this change soon.

Reply:
Fixed in eff8a6f30c9008368b2a69ea6bbd39dd88d011c7.
.